signac/_synced_collections/backends/collection_json.py

```diff
@@ -145,10 +158,17 @@ class JSONCollection(SyncedCollection):
     \*\*kwargs :
         Keyword arguments forwarded to parent constructors.
 
+    Warnings
+    --------
+    This class is _not_ thread safe on Windows. It relies on ``os.replace`` for
+    atomic file writes, and that method can fail in multithreaded situations if
+    open handles exist to the destination file withiin the same process on a
+    different thread. See https://bugs.python.org/issue46003 for more
+    information.
     """
 
     _backend = __name__  # type: ignore
-    _supports_threading = True
+    _supports_threading = _IsWindows()  # type: ignore
```

Comment:
```suggestion
    _supports_threading = not _IsWindows()  # type: ignore
```

Reply:
I think I've been spending too much time in compiled languages and so I created `_IsWindows` just to have something that would be very obviously something evaluated at "runtime". In practice I don't know that this adds any extra clarity and just increases boilerplate. What if I just inlined
```
_supports_threading = not ( sys.platform.startswith("win32") or sys.platform.startswith("cygin"))
```
Then your previous two comments also become moot.

Reply:
I think having a dedicated private constant for this makes sense and makes the code more readable and overall more consistent. We could reuse that variable also in the tests, no?

Reply:
We could. I wouldn't want to leak a constant like that into our public API, but that's the sort of gray area where I would be OK with importing internals into tests I suppose.

Reply:
Done.